osu.Game/Screens/Edit/GameplayTest/EditorPlayer.cs

```diff
@@ -181,19 +185,44 @@ private void toggleAutoplay()
                 if (autoplay == null)
                     return;
 
-                var score = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, [autoplay]);
+                // stores time when autoplay starts
+                previousAutoTime = GameplayClockContainer.CurrentTime;
 
-                // remove past frames to prevent replay frame handler from seeking back to start in an attempt to play back the entirety of the replay.
-                score.Replay.Frames.RemoveAll(f => f.Time <= GameplayClockContainer.CurrentTime);
+                var score = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, new[] { autoplay });
+                allAutoFrames = score.Replay.Frames;
 
                 DrawableRuleset.SetReplayScore(score);
+                if (DrawableRuleset.ReplayScore != null)
+                    DrawableRuleset.ReplayScore.Replay.Frames.RemoveAll(f => f.Time < GameplayClockContainer.CurrentTime);
                 // Without this schedule, the `GlobalCursorDisplay.Update()` machinery will fade the gameplay cursor out, but we still want it to show.
                 Schedule(() => DrawableRuleset.Cursor?.Show());
             }
             else
                 DrawableRuleset.SetReplayScore(null);
         }
 
+        protected override void Update()
+        {
+            // updates if there's a replay (autoplay)
+            if (DrawableRuleset.ReplayScore != null)
+            {
+                // compare currentTime with previousTime
+                if (GameplayClockContainer.CurrentTime < previousAutoTime)
+                {
+                    var drawnFrames = DrawableRuleset.ReplayScore;
+
+                    // find the frames from the original replay data that should be present based on the current time
+                    var missingFrames = allAutoFrames.Where(f => f.Time >= GameplayClockContainer.CurrentTime && f.Time <= previousAutoTime).ToList();
+                    previousAutoTime = GameplayClockContainer.CurrentTime;
+
+                    // add the missing frames back
+                    drawnFrames.Replay.Frames.AddRange(missingFrames);
+
+                    DrawableRuleset.SetReplayScore(drawnFrames);
```

Comment:
All of this kinda sucks.

- Having to refilter the entirety of the autoplay replay...
- ...the frames are added using `AddRange()` which means they're probably going to get added to the *end* of this list which means something is going to have to re-sort them for proper playback...
- ..and a `SetReplayScore` call using _the same exact score as before but mutated_...

and all of the above, _every_ frame while the game is seeking back, at that. I dunno. Maybe some entirely new approach needs to be tried because this just feels way too ugly to live even for workaround standards.

Reply:
I could sort it, but I think approaching this differently will be better for the long run. I stated in #30208 that this simply a fix for a surface level problem.

Reply:
Can we not just regenerate the whole autoplay?

Reply:
Hmm I think I misspoke. Just reconstructing the full frames list rather than doing `AddRange` insertion (should reduce the complexity slightly?)

Reply:
so instead of adding frames, we just take the full list of frames and remove the section?